XenAdmin/Wizards/ImportWizard/ImportSelectNetworkPage.cs

```diff
@@ -51,6 +56,34 @@ protected override bool ImplementsIsDirty()
             return true;
         }
 
+        protected override bool AllowSriovNetwork(XenAPI.Network network, string sysId)
+        {
+            var vhs = OVF.FindVirtualHardwareSectionByAffinity(SelectedOvfEnvelope, sysId, "xen");
+            var data = vhs.VirtualSystemOtherConfigurationData;
+
+            foreach (var s in data)
+            {
+                if (s.Name == "recommendations")
+                {
+                    try
+                    {
+                        XmlDocument xdRecommendations = new XmlDocument();
+                        xdRecommendations.LoadXml(s.Value.Value);
+
+                        XmlNode xn = xdRecommendations.SelectSingleNode(@"restrictions/restriction[@field='allow-network-sriov']");
+                        if (xn != null && xn.Attributes != null)
+                            return Convert.ToInt32(xn.Attributes["value"].Value) != 0;
+                    }
+                    catch
+                    {
+                        return false;
+                    }                    
+                }

```

Comment:
We should exit the foreach loop here if the "recommendation" has been found (and we didn't hit any the two `return`s above) , as there shouldn't be another instance of the "recommendation" data